### ISSUE TITLE
Commit 7680c53 breaks wasm32-emscripten

### ISSUE THREAD

**Issue by floooh** [ziglang/zig#19019]
### Zig Version

0.12.0-dev.2817+7461309b7

TL;DR: Link to commit and file `lib/std/os/wasi.zig`:

https://github.com/ziglang/zig/commit/7680c5330cbc9141b9a5444e30c512b6068ab50d#diff-58f9f43df8e05982774a64ed18397f796f0594db0d9e0290fa0fd738031b9071

...the removed line 283 causes the regression.

### Steps to Reproduce and Observed Behavior

```
git clone https://github.com/floooh/sokol-zig
cd sokol-zig
zig build -Dtarget=wasm32-emscripten
```
(will take a little bit because it pulls down the Emscripten SDK => might make sense to clear your global zig cache afterwards, sorry for the inconvenience)

Build fails with:

```
/Users/floh/.zvm/master/lib/std/os/emscripten.zig:158:32: error: root struct of file 'os.wasi' has no member named 'E'
    SUCCESS = @intFromEnum(wasi.E.SUCCESS),
```

The reason is that commit 7680c5330cbc9141b9a5444e30c512b6068ab50d removed a type alias in `lib/std/os/wasi.zig`:

```
pub const E = errno_t;
```

...but this alias `E` is still used in `lib/std/os/emscripten.zig`:

https://github.com/ziglang/zig/blob/556db2ca36afccd623d80b620139a1428c7b65fa/lib/std/os/emscripten.zig#L157-L236

### Expected Behavior

`wasm32-emscripten` should build :)

**Comment by floooh** [comment]
PS: I can provide a PR but not sure if the preferred solution is to add `E` back to wasi.zig, or use `wasi.errno_t` instead in emscripten.zig (or maybe decouple emscripten.zig completely from wasi.zig?)

**Comment by andrewrk** [comment]
It was a mistake to put `E` in std.os.wasi since that's a libc namespace, not a WASI namespace. So, the fix is to use `std.c.E` instead. 

I'm not sure why os/emscripten.zig exists. As far as I'm aware these are 100% C APIs and should go into std/c/emscripten.zig instead.

And then to fix the issue, in that file, the fix to change `wasi.E` to `c.E`.

**Comment by floooh** [comment]
Oki doki, thanks for the info! I'll try to work on a PR tomorrow.

**Comment by floooh** [comment]
Minimally intrusive PR (just the changes to make wasm32-emscripten build again): https://github.com/ziglang/zig/pull/19030
